apps/scan/backend/src/app_export.test.ts

```diff
@@ -162,6 +162,56 @@ test('continuous CVR export, including polls closing, followed by a full export'
   );
 });
 
+test('continuous CVR export with a mode switch in between', async () => {
```

Comment:
This test covers "Edge case 2" in the PR description. I'd ideally add a test case for "Edge case 1," too. But recreating the relevant race condition would require lots of mocking / tricky timing, so I decided to punt.